### PR TITLE
Fix flaky tests

### PR DESCRIPTION
**Pull request by AbdulrhmnGhanem**

### CODE REVIEW

**Comment by AbdulrhmnGhanem** [comment]
The last 6 runs got stable with only 2 flaky tests, so I focus on those.
image
Also, I think if we are going to consider flaky tests a failure case, then we should remove `retries`.

**Comment by AbdulrhmnGhanem** [comment]
> Also, I think if we are going to consider flaky tests a failure case, then we should remove `retries`.

@kasbah, probably you haven't noticed this.

**Comment by kasbah** [comment]
Remove `retries` from where? Not sure what you are talking about or what I would have been able to notice.

**Comment by AbdulrhmnGhanem** [comment]
> Remove `retries` from where? Not sure what you are talking about or what I would have been able to notice.

Cypress retries, here:
https://github.com/kitspace/kitspace-v2/blob/7f8629e06ea05fb2865c1785a1dba8167ef17983/frontend/cypress.json#L10-L13

A test marked `flaky` by cypress when it runs it, it fails, it retries it again as many times as specified in `cypress.json` if all retries fail then this is marked as a failure not `flaky`.

**Comment by kasbah** [comment]
Ok, seems like we should still have some retries though. Without retries, how would it identify a flaky test? Anyway, I'll let you handle it.

**Comment by kasbah** [comment]
You think this is ready?

**Comment by AbdulrhmnGhanem** [comment]
> You think this is ready?

Yes. Though I think we shouldn't consider flaky tests a failure in the future. This took too much time and effort yet providing little value.

File: frontend/cypress/integration/updateProject.spec.js [hunk 1]
```diff
@@ -1,136 +1,5 @@
...
-    cy.get('[data-cy=update-form-submit]').click()
-
-    // Should redirect to the new update page
-    cy.url().should('contain', `/${username}/${newName}`)
-
-    // `example3.png` should be in the preview
-    cy.get('[data-cy=file-name]').contains('example3.png')
-  })
-})
-
```
Comment (kasbah) [inline frontend/cypress/integration/updateProject.spec.js:133]: How come this whole test is being deleted?

**Comment by kasbah** [comment]
> Yes. Though I think we shouldn't consider flaky tests a failure in the future. This took too much time and effort yet providing little value.

I disagree, we need to be able to trust the results from our test suite. Having a flaky test is worse than no test.

File: frontend/cypress/integration/updateProject.spec.js (hunk 1 above)
Comment (AbdulrhmnGhanem) [reply frontend/cypress/integration/updateProject.spec.js:133]: This is a mistake! I looked at the history for this file, I deleted it in an unrelated commit to this file, will roll that part back.

File: .github/workflows/docker.yml [hunk 2]
```diff
@@ -171,11 +171,6 @@ jobs:
     runs-on: ubuntu-20.04
     needs:
       - build
-    strategy:
-      fail-fast: false
-      matrix:
-        # run copies of the current job in parallel
-        containers: [1, 2, 3, 4]
```
Comment (kasbah) [inline .github/workflows/docker.yml:178]: Why are you getting rid of the parallel testing? Seems to speed things up quite a bit no?

Comment (AbdulrhmnGhanem) [reply .github/workflows/docker.yml:178]: It seems it increases the flakiness of the test, probably due to the start-up time of the frontend. Even looking at the cypress dashboard didn't help, the recording starts at random steps. I leave this for a different pr.

**Pull request merged**